charts/seed-operatingsystemconfig/original/templates/containerd/containerd-require-config-generator.yaml

```diff
@@ -0,0 +1,12 @@
+{{- define "containerd-require-config-generator" -}}
+- path: /etc/systemd/system/containerd.service.d/10-require-config-generator.conf
+  permissions: 0644
+  content:
+    inline:
+      encoding: ""
+      data: |
+        [Unit]
+        After=containerd-config-generator.service
+        Requires=containerd-config-generator.service
+        PartOf=containerd-config-generator.service
```

Comment:
do your really need all three `After,Requires,PartOf` here?

Reply:
After is required for the timely ordering. So that the generator with type=oneshot completes before the containerd.service is started/restarted.

PartOf is needed so that the containerd service is restarted when the generator service is (also stopped but it is a type=oneshot service so its not relevant). 

Requires is needed so that the generator service is started when the containerd service is started.  Also instead of 'Wants' as this lets the containerd.service fail if the config generation.service failed to start. 

This is my current understanding. During my tests that worked, however I might be missing something here.  Would appreciate a verification.

Reply:
Again a suggestion to avoid redundancy, based on my understanding `Requires` includes all cases but `PartOf` only is about starting and stopping, but I could be wrong.

Reply:
I tested it and based on that PartOf can be removed. Thanks!